akka-http-core/src/main/scala/akka/http/impl/engine/rendering/HttpResponseRendererFactory.scala

```diff
@@ -79,29 +81,39 @@ private[http] class HttpResponseRendererFactory(
               case Strict(outElement) =>
                 push(out, outElement)
                 if (close) completeStage()
-              case Streamed(outStream) => transfer(outStream)
+              case Streamed(outStream) =>
+                try transfer(outStream)
+                catch {
+                  case NonFatal(e) =>
+                    transferring = false
+                    log.error(e, s"Rendering of response failed because response entity stream materialization failed with '${e.getMessage}'. Sending out 500 response instead.")
+                    push(out, render(ResponseRenderingContext(HttpResponse(500, entity = "OOPS!"))).asInstanceOf[Strict].bytes)
```

Comment:
:smile:

Reply:
Let's use the default message there as well ;)

Updated.